File: isMonotonic.java
class Solution2 {
    public static void main(String[] args) {
      Solution2 ins = new Solution2();
      int[] arr = {1,2,3,3};
      System.out.println(ins.isMonotonic(arr) == true);
      int[] arr2 = {6,5,4,4};
      System.out.println(ins.isMonotonic(arr2) == true);
      int[] arr3 = {1,3,2,2};
      System.out.println(ins.isMonotonic(arr3) == false);
      int[] arr4 = {3,3,2,2};
      System.out.println(ins.isMonotonic(arr4) == true);
      int[] arr5 = {-5963,-123,8,61,4649};
      System.out.println(ins.isMonotonic(arr5) == true);
      int[] arr6 = {4,4,6,6};
      System.out.println(ins.isMonotonic(arr6) == true);
      int[] arr7 = {666,-999};
      System.out.println(ins.isMonotonic(arr7) == true);
      int[] arr8 = {0,1,8,100};
      System.out.println(ins.isMonotonic(arr8) == true);
      int[] arr9 = {8,3,0,-1,-4,-8};
      System.out.println(ins.isMonotonic(arr9) == true);
      int[] arr10 = {5,5,3,4};
      System.out.println(ins.isMonotonic(arr10) == false);
      int[] arr11 = {5,3,4,4};
      System.out.println(ins.isMonotonic(arr11) == false);
      int[] arr12 = {5,3,4};
      System.out.println(ins.isMonotonic(arr12) == false);
      int[] arr13 = {10,9,-2,7};
      System.out.println(ins.isMonotonic(arr13) == false);
      int[] arr14 = {-321,0,8,-7};
      System.out.println(ins.isMonotonic(arr14) == false);
      int[] arr15 = {-99,0,8,7};
      System.out.println(ins.isMonotonic(arr15) == false);
      int[] arr16 = {0,0,0,0,0,0,0};
      System.out.println(ins.isMonotonic(arr16) == true);
      int[] arr17 = {0};
      System.out.println(ins.isMonotonic(arr17) == true);
      System.out.println("すべてTrueならコンプリート。本番テストへ");
    }

    public boolean isMonotonic(int[] nums) {
      // ここにコードを書いてください
      
      //カウントするための変数
      int count = 0;
      int a = 1;
      int preValue = 0;

      //
      //boolean xxx = nums;
      
      //文字列の文字数分まで繰り返すループ文
      for(int i = 0 ; i<nums.length; i++){
        //charCount = frontAndRearSpacesExcluded.charAt(count);
        System.out.println(nums[i]);
        count = a++;
        if(count >= 1){

        }
        if(count >= 0){
          preValue = nums[i];
        }
      }
        return 2 > 1;
    }
}

File: isMonotonic.py
from typing import *

class Solution:
    def isMonotonic(self, nums: List[int]) -> bool:
        # ここにコードを書いてください

##################################################

        # preValue を用意するための分岐用変数
        count = 0
        #ひとつ前の値
        preValue = 0
        # preValue と i を比較して preValue が i 未満だった場合のルート分岐変数
        route1 = 0
        # preValue と i を比較して preValue が i より上だった場合のルート分岐変数
        route2 = 0

        # nums の数値をひとつずつ i に代入して繰り返す
        for i in nums :
            count += 1
            """
            print("カウント：",count)
            print("nums:",i)
            print("route",route1,route2)
            """

            #ルート分岐
            if route1 == 1 or route2 == 2 :

                # route1 : preValue が i 以下なら何もしない、それ以外はFalseを引数に返す
                if route1 == 1:
                    if preValue <= i:
                        pass
                    else:
                        return False

                # route2 : preValue が i 以上なら何もしない、それ以外はFalseを引数に返す
                elif route2 == 2:
                    if preValue >= i:
                        pass
                    else:
                        return False

            # count が 2 以上だった場合
            if count >= 2:
                # preValue が i より小さい場合、 route1 に 1 を代入する
                if preValue < i:
                    route1 = 1

                # preValue が i より大きい場合、 route2 に 2 を代入する
                elif preValue > i:
                    route2 = 2

            # count が 1 以上だった場合、 i を preValue に代入する
            if count >= 1:
                preValue = i

        """
        print(i)
        print("--------------")
        """
        # True を引数に返す
        return True

##################################################

print("スタート")
assert Solution().isMonotonic([1,2,2,3]) == True
assert Solution().isMonotonic([2,2,3,3]) == True
assert Solution().isMonotonic([-5963,-123,8,61,4649]) == True
assert Solution().isMonotonic([6,5,4,4]) == True
assert Solution().isMonotonic([666,-999]) == True
assert Solution().isMonotonic([0.8,1.29,80.5,100]) == True
assert Solution().isMonotonic([8,2.3,0.55,-1,-3.4,-8]) == True
assert Solution().isMonotonic([1,3,2]) == False
assert Solution().isMonotonic([5,5,3,4]) == False
assert Solution().isMonotonic([5,3,4,4]) == False
assert Solution().isMonotonic([5,3,4]) == False
assert Solution().isMonotonic([10,9,-2,7]) == False
assert Solution().isMonotonic([-321,0,8,-7]) == False
assert Solution().isMonotonic([-99,0,0.8,0.7]) == False
assert Solution().isMonotonic([0]) == True
assert Solution().isMonotonic([0,0,0,0,0,0,0]) == True
print("コンプリート!本番テストへ")
